Fixes Bug of CreateGameDialog

hammad-521 opened the pull request:
<!-- Thanks for contributing to Cuttle! 🎉 -->

## Issue number
- Resolves #518 

## Please check the following

- [x] Do the tests still pass? 
- [x] Is the code formatted properly? 


## Description:

> Replaced the form around the switch and scoring dialog with a  div, ensuring they appear on the same line.
> Moved the v-form wrapper to include both the switch and the text field within the dialog body.
> The submit button will trigger the form submission using the submitNewGame method.

Haviles04 commented on src/routes/home/components/CreateGameDialog.vue:
```diff
@@ -29,20 +29,20 @@
-          color="surface-2"
-        />
-      </form>
       <v-form @submit.prevent="submitNewGame">
```
I believe we wanted to wrap the whole #body with the v-form, so from lines 20 - 56

hammad-521 replied:
I've made the changes you requested, wrapping the entire #body with the <v-form> tag as per the PR description. The commit has been pushed, and everything should now be in line with the requirements.

Please let me know if you need any further adjustments.

Thanks,
Hammad

Haviles04 reviewed (commented):
Thanks for contributing! This looks good, one small comment about wrapping the whole contents as per the PR description 👍

Haviles04 commented on src/routes/home/components/CreateGameDialog.vue:
```diff
@@ -18,31 +18,32 @@
-        />
-      </form>
       <v-form @submit.prevent="submitNewGame">
+        <!-- V form now wraps the whole #body -->
```
This comment is unnecessary

hammad-521 replied:
I have removed this comment in latest commit.

Haviles04 reviewed (commented):
This is looking good, good job!

Curious if we should include the #actions template in the form as well

@itsalaidbacklife thoughts?

Haviles04 commented:
> This is looking good, good job!
> 
> Curious if we should include the #actions template in the form as well
> 
> @itsalaidbacklife thoughts?

After some discussion I think the move here is to add an `id` to the `form` and add a `form=id` attribute to the submit button, we could then remove the `@click` on the button

hammad-521 commented:
@Haviles04 Thank you for the feedback and discussion. I have implemented the suggested changes:

- Added an id attribute to the v-form element: id="create-game-form".
- Added a form attribute to the submit button: form="create-game-form".
- Removed the @click handler from the submit button.

I've also removed the unnecessary comment about wrapping the whole #body with v-form.  

Please review the latest commit and let me know if any further adjustments are needed.

Haviles04 reviewed (approved):
Great Work! Thank you for your contribution!
